### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by janisozaur**
Includes (mostly) type fixes, dead code removal, signed vs unsigned
comparison fixes, memory leaks removal.

### CODE REVIEW

File: src/object_list.c [hunk 1]
```diff
@@ -473,14 +473,14 @@ void object_create_identifier_name(uint8* string_buffer, rct_object_entry* objec
 
 	for (uint8 i = 0; i < 4; ++i){
 		uint8 flag_part = (object->flags >> (i * 8)) & 0xFF;
-		*string_buffer++ = RCT2_ADDRESS(0x0098DA64, uint8)[flag_part >> 4];
-		*string_buffer++ = RCT2_ADDRESS(0x0098DA64, uint8)[flag_part & 0xF];
+		*string_buffer++ = RCT2_ADDRESS(0x0098DA64, char)[flag_part >> 4];
```
Comment (warmwaffles) [inline src/object_list.c]: why the switch back to char from uint8?

**Comment by warmwaffles** [comment]
Could you split the whitespace removals up from the actual code changing? It'd be easier to revert a single commit than have to rummage through what changed.

File: src/object_list.c (hunk 1 above)
Comment (janisozaur) [inline src/object_list.c]: Because signedness of `char` is undefined by standard. See http://stackoverflow.com/a/2055015

**Comment by janisozaur** [comment]
I fail to see how this is a problem. All the utilities support whitespace stripping, at runtime.

File: src/object_list.c (hunk 1 above)
Comment (IntelOrca) [inline src/object_list.c]: Seems like a good reason to never use it - always prefer sint8 or uint8 or char8.

**Comment by IntelOrca** [comment]
@janisozaur If you do a blame your name appears everywhere for just removing whitespace.

**Comment by janisozaur** [comment]
@IntelOrca yeah, sure, until you do `git blame -w` https://git-scm.com/docs/git-blame

**Comment by IntelOrca** [comment]
w=1 doesn't seem to work on github.

**Comment by janisozaur** [comment]
I'd like to get rid of all trailing whitespace, would you accept a single commit fixing all of that and only that for whole project?

**Comment by IntelOrca** [comment]
yes

**Comment by warmwaffles** [comment]
@janisozaur I'm not worried about the blaming, but I have had instances where I need to revert a commit but I don't want to revert white space changes or want to revert code changes. I guess right now it's not an issue but it is a good habit to have.

**Comment by janisozaur** [comment]
Rebased on top of whitespace cleanup

**Comment by IntelOrca** [comment]
@janisozaur Will you be changing the chars to sint8 or uint8? If you said the sign is undefined then we should avoid using them.

**Comment by janisozaur** [comment]
I guess I can do that in the evening, but all the `strcpy` and others use `char *`. That shouldn't affect results of these functions, just a cast to silence a possible warning, so I'd say it's fine as it is anyway.

Would you like me to change them?

**Comment by IntelOrca** [comment]
No its fine, they will all use our new string functions (char8 or utf8) when we go C++ anyway.

**Pull request merged**